src/theory/sets/theory_sets_private.cpp

```diff
@@ -985,9 +985,20 @@ void TheorySetsPrivate::computeCareGraph()
       {
         Trace("sets-cg-debug") << "...build for " << f1 << std::endl;
         Assert(d_equalityEngine->hasTerm(f1));
-        // break into index based on operator, and type of first argument (since
-        // some operators are parametric)
-        TypeNode tn = f1[0].getType();
+        // break into index based on operator, and the type of the element
+        // type of the proper set, which notice must be safe wrt subtyping.
```

Comment:
`,and the type of the element type of the proper set,`
Do you mean the `get the type of the element from the type of the proper set`?

Reply:
No, this is the purpose of the fix: we need to consider the element type of the set, not the type of the element in the set.